### PR TITLE
Issue: #10 - Added typescript typings.

### PR DESCRIPTION
**Pull request by gavinharriscolab**
Provide Typescript Typings via index.d.ts for the Library. I think I have
covered everything but will enhance as needed.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -16,6 +16,11 @@ declare module "react-native-get-location" {
     timeout: number;
   }
 
+  export class LocationError extends Error {
+    name: string;
+    code: string;
```
Comment (douglasjunior) [inline index.d.ts:21]: I think that `code` must have a type, like:

```ts
type errorCodeType = 'CANCELLED' | 'UNAVAILABLE' | 'TIMEOUT' | 'UNAUTHORIZED';
```

Then:
```ts
export class LocationError extends Error {
    message: string;
    code: errorCodeType;
}
```

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  export class LocationError extends Error {
+    name: string;
+    code: string;
+  }
+
+  export function getCurrentPosition(
+    options: GetCurrentPositionOptions
+  ): Promise<Location>;
```
Comment (douglasjunior) [inline index.d.ts:26]: Here we need to define the error?

```suggestion
  ): Promise<Location, LocationError>;
```

Comment (gavinharriscolab) [reply index.d.ts:26]: Going to go check, if the error is returned (not thrown) by the `getCurrentPosition` function then yes I should have put it in the Promise. Let me go check.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  }
+
+  export type LocationErrorCode =
+    | "CANCELLED"
+    | "UNAVAILABLE"
+    | "TIMEOUT"
+    | "UNAUTHORIZED";
+
+  export class LocationError extends Error {
+    name: string;
```
Comment (douglasjunior) [inline index.d.ts:26]: Here we have a `message? : string` instead of `name`.

File: index.d.ts (hunk 2 above)
Comment (douglasjunior) [reply index.d.ts:32]: The error is thrown.

File: index.d.ts (hunk 3 above)
Comment (gavinharriscolab) [reply index.d.ts]: I had a quick look at Error and it already exposes: 
`message: string, name: string`
So I have removed the `name: string` override.

**Comment by douglasjunior** [comment]
@gavinharriscolab is something still missing?

**Pull request merged**